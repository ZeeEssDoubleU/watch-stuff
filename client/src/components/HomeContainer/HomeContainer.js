import React, { Component } from "react";

import VideoGrid from "../VideoGrid/VideoGrid";
import SideBarContainer from '../SideBarContainer/SideBarContainer';
import './HomeContainer.scss';

class HomeContainer extends Component {
	render() {
		return (
			<>
				<SideBarContainer />
				<div className="HomeContainer">
					<div className="responsive-video-grid-container">
						<VideoGrid title="Trending" />
						<VideoGrid title="Autos & Vehicles" hideDivider={true} />
					</div>
				</div>
			</>
		);
	}
}

export default HomeContainer;
