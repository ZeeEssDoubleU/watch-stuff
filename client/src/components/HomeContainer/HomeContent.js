import React, { Component } from "react";
import { connect } from "react-redux";

import "./HomeContent.scss";
import VideoGrid from "../VideoGrid/VideoGrid";
import InfiniteScroll from "../InfiniteScroll/InfiniteScroll";

import {
	selector_mostPopularVideos,
	selector_mostPopularVideosByCategory,
	selector_mostPopularVideosByCategoryLoaded,
	selector_mostPopularVideosByCategoryLength,
} from "../../store/reducers/videos";

class HomeContent extends Component {
	constructor(props) {
		super(props);
		this.state = { lazyLoadIndex: 0 };
	}

	componentDidUpdate() {
		let homeContentHeight =
			document.querySelector(".home-content").offsetHeight -
			document.querySelector(".loader-container").offsetHeight;
		if (
			this.props.videosByCategoryLoaded &&
			homeContentHeight < window.innerHeight
		) {
			this.lazyLoadVideoCategories();
		}
	}

	lazyLoadVideoCategories = () => {
		if (
			this.props.videosByCategoryLoaded &&
			this.state.lazyLoadIndex < this.props.videosByCategoryLength
		) {
			this.setState({ lazyLoadIndex: this.state.lazyLoadIndex + 1 });
		}
	};

	shouldShowLoader = () => {
		// if vidoesByCategory loaded, return true if lazyLoadIndex < total categories
		return this.props.videosByCategoryLoaded &&
			this.state.lazyLoadIndex < this.props.videosByCategoryLength
			? true
			: false;
	};

	render() {
		const categoryNames = Object.keys(this.props.videosByCategory);
		const categoryGrids = categoryNames
			.slice(0, this.state.lazyLoadIndex)
			.map((categoryName, index) => (
				<VideoGrid
					title={categoryName}
					key={categoryName}
					videos={this.props.videosByCategory[categoryName]}
					hideDivider={index === categoryNames.length - 1}
				/>
			));

		return (
			<div className="home-content">
				<div className="responsive-video-grid-container">
					<InfiniteScroll
						lazyLoadCallback={this.lazyLoadVideoCategories}
						showLoader={this.shouldShowLoader()}>
						<VideoGrid title="Trending" videos={this.props.mostPopular} />
						{categoryGrids}
					</InfiniteScroll>
				</div>
			</div>
		);
	}
}

const mapStateToProps = state => ({
	mostPopular: selector_mostPopularVideos(state),
	videosByCategory: selector_mostPopularVideosByCategory(state),
	videosByCategoryLoaded: selector_mostPopularVideosByCategoryLoaded(state),
	videosByCategoryLength: selector_mostPopularVideosByCategoryLength(state),
});

export default connect(
	mapStateToProps,
	null,
)(HomeContent);
