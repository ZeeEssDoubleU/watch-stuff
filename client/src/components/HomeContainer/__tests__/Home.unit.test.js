import React from "react";
import { shallow } from "enzyme";

import HomeContainer from "../HomeContainer";

test("renders HomeContainer", () => {
   const wrapper = shallow(<HomeContainer />);
   expect(wrapper).toMatchSnapshot();
});
