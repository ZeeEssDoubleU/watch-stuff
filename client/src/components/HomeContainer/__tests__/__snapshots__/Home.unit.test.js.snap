// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HomeContainer 1`] = `
<Fragment>
  <SideBarContainer />
  <div
    className="HomeContainer"
  >
    <div
      className="responsive-video-grid-container"
    >
      <VideoGrid
        title="Trending"
      />
      <VideoGrid
        hideDivider={true}
        title="Autos & Vehicles"
      />
    </div>
  </div>
</Fragment>
`;
